hw3/01-characters.js

```diff
@@ -1,4 +1,88 @@
 // url for the Thrones API
 const url = 'https://thronesapi.com/api/v2/Characters';
+const charactersContainer = document.querySelector('.characters-container');
 
-fetch(url);
+const createRow = function createGenericImgRowForDOM(rowIndex) {
+  const newRow = document.createElement('div');
+
+  newRow.className = 'row row-cols-2 row-cols-lg-4 align-self-center';
+  newRow.id = `row-${rowIndex}`;
+
+  charactersContainer.append(newRow);
+};
+
+const createImg = function createGenericImgForDOM(rowIndex, imgIndex) {
+  const rowContainer = document.getElementById(`row-${rowIndex}`);
+  const newCol = document.createElement('div');
+  const newFigure = document.createElement('figure');
+  const newImg = document.createElement('img');
+  const newFigCaption = document.createElement('figcaption');
+  const newSmall = document.createElement('small');
+
+  newCol.className = 'col-auto p-1 character-container';
+
+  newFigure.id = `img-${imgIndex}`;
+
+  newImg.className = 'character-img';
+  newImg.style.width = '250px';
+  newImg.style.height = '300px';
```

Comment:
I would recommend setting the width and height attributes in the style.css file. To clean up the code a bit and separation of concerns since you are using bootstrap as well!

Reply:
That's a good point I'll move that to css, thanks!